### PR TITLE
Added RunMsMarco V2.1, Added RunBeir

### PR DESCRIPTION
**Pull request by wu-ming233**
- Added command-line option for RunMsMarco repro to support V2.1 and V2.1 segmented indexes.
- Added a new repro for RunBeir with output format following the instructions in [fatjar-regressions-v0.35.1.md](https://github.com/castorini/anserini/blob/master/docs/fatjar-regressions-v0.35.1.md).
- Removed some minor dead code.

### CODE REVIEW

**Comment by wu-ming233** [comment]
Refs #2473

File: src/main/java/io/anserini/reproduce/RunMsMarco.java [hunk 1]
```diff
@@ -39,66 +33,81 @@ public class RunMsMarco {
...
+
   public static void main(String[] args) throws Exception {
-    final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-    try (InputStream inputStream = RunMsMarco.class.getClassLoader()
-        .getResourceAsStream("reproduce/msmarco-v1-passage.yaml")) {
-      if (inputStream == null) {
-        throw new FileNotFoundException("Resource file not found. Please check the file path.");
+
+    // check for cmd option
+    if (args.length >= 2 && "-v".equals(args[0])) {
```
Comment (lintool) [inline src/main/java/io/anserini/reproduce/RunMsMarco.java:41]: Can you parse args in a manner that's consistent with everything else in repo? E.g., https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/index/IndexCollection.java#L272

File: src/main/java/io/anserini/reproduce/RunMsMarco.java [hunk 2]
```diff
@@ -39,66 +33,81 @@ public class RunMsMarco {
...
+      String collectionVersion = args[1];
+      switch (collectionVersion) {
+          case "1":
+              COLLECTION = "msmarco-v1-passage";
+              break;
+          case "2":
+              COLLECTION = "msmarco-v2.1-doc";
```
Comment (lintool) [inline src/main/java/io/anserini/reproduce/RunMsMarco.java:48]: Ultimately, we want something like https://castorini.github.io/pyserini/2cr/msmarco-v2-doc.html

So it's okay with put doc and doc segmented together.

File: src/main/resources/reproduce/msmarco-v2.1-doc.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+conditions:
```
Comment (lintool) [inline src/main/resources/reproduce/msmarco-v2.1-doc.yaml:1]: Combine doc with doc-segmented?

**Review by lintool — CHANGES_REQUESTED** [review]
Lots of code duplication... is there any low-hanging fruit in terms of abstracting out common code?

File: src/main/resources/reproduce/msmarco-v2.1-doc.yaml (hunk 3 above)
Comment (wu-ming233) [reply src/main/resources/reproduce/msmarco-v2.1-doc.yaml:1]: While coding I was thinking how RunBeir could just simply be another command line option in RunMsMarco, due to the uniform interface in the yaml files. Would it make sense if we combine them into one script, called "RunRepro" or something similar, and specify the different regressions in yaml files?

Comment (lintool) [reply src/main/resources/reproduce/msmarco-v2.1-doc.yaml:1]: Currently, in Pyserini, each corpus is a separate file: https://github.com/castorini/pyserini/tree/master/pyserini/2cr

We're modelling after that repo as a start, but can probably further combine later?

For now, let's have MS MARCO and BEIR as separate classes, and then maybe circle back?

**Comment by wu-ming233** [comment]
Refactored RunMsMarco and RunBeir; put common code in a new "RunRepro".
Note that src/test/java/io/anserini/doc/GenerateReproductionDocsTest.java was modified to use an updated variable name, though the semantics remained.

**Comment by lintool** [comment]
Hi @wu-ming233 can you resolve conflict and merge in #2476 ? Thanks!

**Comment by wu-ming233** [comment]
Merged with #2476 and updated the corresponding yaml's for other regressions as well, adhering to #2476.

**Review by lintool — APPROVED** [review]

**Pull request merged**